protocol/rtmp/src/session/server_session.rs

```diff
@@ -672,8 +677,16 @@ impl ServerSession {
             }
         };
 
-        (self.stream_name, self.query) =
-            RtmpUrlParser::parse_stream_name_with_query(&stream_name_with_query);
+        let mut url = RtmpUrlParser::new(
```

Comment:
the stream_name_with_query with empty string is a special case, so I think the following logic is better:

        if !stream_name_with_query.is_empty() {
            (self.stream_name, self.query) =
                RtmpUrlParser::parse_stream_name_with_query(&stream_name_with_query);
        } else {
            let mut url = RtmpUrlParser::new(
                self.connect_properties
                    .swf_url
                    .clone()
                    .unwrap_or("".to_string()),
            );
            _ = url.parse_url();

            self.stream_name = url.stream_name;
            self.query = url.query;
        }